engine/framework/base_jobmaster.go

```diff
@@ -201,13 +201,15 @@ func (d *DefaultBaseJobMaster) Init(ctx context.Context) error {
 
 	if isFirstStartUp {
 		if err := d.impl.InitImpl(ctx); err != nil {
+			d.errCenter.OnError(err)
 			return errors.Trace(err)
 		}
 		if err := d.master.markStateInMetadata(ctx, frameModel.MasterStateInit); err != nil {
 			return errors.Trace(err)
 		}
 	} else {
 		if err := d.impl.OnMasterRecovered(ctx); err != nil {
+			d.errCenter.OnError(err)
```

Comment:
1. Would it be better to set the `errCenter` in `eventLoop`? Or we may lost some error message if we forget to set the `errCenter`.
2. Please add some comment here if we have specific strategy when record the error message.

Reply:
1. ErrCenter is logic oriented, instead of runtime work, it is better to record error in implement code, rather than in runtime layer.
2. Comments added. Only record error message when business API returns error.

Reply:
PTAL @maxshuang